common/app/uiComponent/TestComponent.scala

```diff
@@ -3,5 +3,5 @@ package uiComponent
 import play.api.libs.json.JsValue
 
 object TestComponent extends UIComponent {
-  override def asJson: Option[JsValue] = None
+  override def state: Option[JsValue] = None
```

Comment:
Its really `props` isn't it? In react state means something different and in the UI part we are receiving props.